test/matcher_combinators/core_test.clj

```diff
@@ -264,3 +264,22 @@
     => [:mismatch (model/->Mismatch matchers [5])]
     (#'core/match-any-order matchers [5] true)
     => [:mismatch (model/->Mismatch matchers [5])]))
+
+(future-fact "reproducing select? bug"
```

Comment:
This future-fact refers to another bug, which will take a bit more thinking to fix.